slackbridge/messages.py

```diff
@@ -71,17 +82,65 @@ def resolve(self):
                 user_bot.leave(channel_name)
             return
 
+    def is_bot_user(self):
+        """Sometimes bot_id is not included and other
+        times it is passed as None. Checks both cases."""
+        return ('bot_id' in self.raw_message and
+                self.raw_message['bot_id'] is not None)
+
     def _change_presence(self, user_bot):
         if self.raw_message['presence'] == 'away':
             user_bot.away('Slack user inactive.')
         elif self.raw_message['presence'] == 'active':
             user_bot.back()
 
-    def _irc_me_action(self, channel_name, user_bot):
+    def _irc_me_action(self, channel_name, user_bot, action):
         user_bot.post_to_irc(
             user_bot.describe,
             '#' + channel_name,
-            self.raw_message['text'],
+            action,
+        )
+
+    def _post_to_fluffy(self, channel_name, user_bot, file_data):
+        # Adapted from https://api.slack.com/tutorials/working-with-files
+        auth = {'Authorization': 'Bearer {}'.format(
+            self.bridge_bot.slack_token
+        )}
+        r = requests.get(
+            file_data['url_private'],
+            headers=auth,
+            stream=True,
+        )
+        if r.status_code != 200:
+            log.err('Could not GET image from: {}'.format(
+                file_data['url_private'],
+            ))
+            return
+
+        # Ensure file has an extension as this is necessary
+        # for fluffy to give a direct link in the browser.
+        filename = file_data['title']
+        if not os.path.splitext(filename)[1]:
+            filename += '.' + file_data['filetype']
+
+        # Decompress file data and upload the file data as
+        # it is streamed.
+        r.raw.decode_content = True
+        r = requests.post(
+            FILEHOST + '/upload',
+            files={'file': (filename, r.raw)},
+            allow_redirects=False,
+        )
```

Comment:
it might make more sense to use the `/upload?json` API? this will also allow you to grab direct image links if you want them, e.g.

```python
import requests
import io

req = requests.post(
    'https://fluffy.cc/upload?json',
    files={'file': ('data', io.StringIO('some text'))},
)
print(req.json()['uploaded_files']['data']['raw'])
```

the json response looks like this:

```json
{
    "redirect": "https://i.fluffy.cc/K33ksRw5fHpdFfxPbrD4CDl95d7cjWkn.html",
    "success": true,
    "uploaded_files": {
        "data": {
            "paste": "https://i.fluffy.cc/phf8kZqBt0F29kqb8c4gq6PxxH4xrlFK.html",
            "raw": "https://i.fluffy.cc/2BZSfnwL2MLSh348tp4C90h608gsBVbB"
        }
    }
}
```